File: sdks/typescript/examples/quickstart.ts
const sdkModuleUrl = new URL("../../dist/index.js", import.meta.url);
const { EchoAgentClient } = (await import(sdkModuleUrl.href)) as typeof import("../src/index.js");

const hostCommand = process.env.ECHO_AGENT_SDK_HOST;
const hostConfig = process.env.ECHO_AGENT_SDK_CONFIG;
if (!hostCommand || !hostConfig) {
  throw new Error("ECHO_AGENT_SDK_HOST and ECHO_AGENT_SDK_CONFIG are required");
}

const sdk = await EchoAgentClient.spawn({
  hostCommand,
  args: ["--config", hostConfig],
});
try {
  const agent = await sdk.createAgent();
  const session = await agent.createSession();
  try {
    const name = await session.invoke<string>("echo_core::agent::Agent::name");
    console.log(name);
  } finally {
    await session.close();
    await agent.close();
  }
} finally {
  await sdk.close();
}

File: sdks/python/examples/quickstart.py
from __future__ import annotations

import asyncio
import os

from echo_agent_sdk import EchoAgentClient


async def main() -> None:
    host = os.environ.get("ECHO_AGENT_SDK_HOST")
    config = os.environ.get("ECHO_AGENT_SDK_CONFIG")
    if not host or not config:
        raise RuntimeError("ECHO_AGENT_SDK_HOST and ECHO_AGENT_SDK_CONFIG are required")
    async with await EchoAgentClient.spawn(host, "--config", config) as sdk:
        agent = await sdk.create_agent()
        try:
            session = await agent.create_session()
            try:
                print(await session.invoke("echo_core::agent::Agent::name"))
            finally:
                await session.close()
        finally:
            await agent.close()


if __name__ == "__main__":
    asyncio.run(main())
